src/index.js

```diff
@@ -274,26 +292,48 @@ const aws = require("aws-sdk");
 aws.config.region = "${this.options.region}";
 const lambda = new aws.Lambda();
 const functionNames = ${JSON.stringify(functionNames)};
+const functionConcurrency = ${JSON.stringify(fullFunctionConcurrency)};
 module.exports.warmUp = async (event, context, callback) => {
   console.log("Warm Up Start");
   const invokes = await Promise.all(functionNames.map(async (functionName) => {
-    const params = {
-      ClientContext: "${Buffer.from(`{"custom":${this.warmup.source}}`).toString('base64')}",
-      FunctionName: functionName,
-      InvocationType: "RequestResponse",
-      LogType: "None",
-      Qualifier: process.env.SERVERLESS_ALIAS || "$LATEST",
-      Payload: '${this.warmup.source}'
-    };
-
-    try {
-      const data = await lambda.invoke(params).promise();
-      console.log(\`Warm Up Invoke Success: \${functionName}\`, data);
-      return true;
-    } catch (e) {
-      console.log(\`Warm Up Invoke Error: \${functionName}\`, e);
-      return false;
+    let concurrency = functionConcurrency[functionName] > 0 ? functionConcurrency[functionName] : 1;
+    let source = ${this.warmup.source};
+    let promises = [];
+    
+    console.log(\`Warming up function: \${functionName} with concurrency: \${concurrency}\`);
+    
+    for (let x = 0; x < concurrency; x++) {
+      source.concurrencyIndex = x;
```

Comment:
According to https://github.com/FidelLimited/serverless-plugin-warmup/pull/81/files#diff-1fdf421c05c1140f6d71444ea2b27638R117, `source` is a string here and not an object.

Reply:
I'll put some more thought into how this is handled.